pkg/model/resource/resource.go

```diff
@@ -78,10 +137,94 @@ func (r Resource) Replacer() *strings.Replacer {
 	var replacements []string
 
 	replacements = append(replacements, wrapKey("group"), r.Group)
-	replacements = append(replacements, wrapKey("group-package-name"), r.GroupPackageName)
 	replacements = append(replacements, wrapKey("version"), r.Version)
 	replacements = append(replacements, wrapKey("kind"), strings.ToLower(r.Kind))
 	replacements = append(replacements, wrapKey("plural"), strings.ToLower(r.Plural))
+	replacements = append(replacements, wrapKey("package-name"), r.PackageName())
 
 	return strings.NewReplacer(replacements...)
 }
+
+// API holds the information related to the golang type definition and the CRD.
+type API struct {
+	// Version holds the CustomResourceDefinition API version used for the resource.
+	Version string `json:"crdVersion,omitempty"`
+
+	// Namespaced is true if the resource is namespaced.
+	Namespaced bool `json:"namespaced,omitempty"`
+}
+
+// update combines fields of the APIs of two resources.
+func (api *API) update(other *API) error {
+	// If other is nil, nothing to merge
+	if other == nil {
+		return nil
+	}
+
+	// Update the version.
+	if other.Version != "" {
+		if api.Version == "" {
+			api.Version = other.Version
+		} else if api.Version != other.Version {
+			return fmt.Errorf("CRD versions do not match")
+		}
+	}
+
+	// Update the namespace.
+	api.Namespaced = api.Namespaced || other.Namespaced
+
+	return nil
+}
+
+// IsEmpty returns if the API's fields all contain zero-values.
+func (api API) IsEmpty() bool {
+	return api.Version == "" && !api.Namespaced
+}
+
+// Webhooks holds the information related to the associated webhooks.
+type Webhooks struct {
+	// Version holds the {Validating,Mutating}WebhookConfiguration API version used for the resource.
+	Version string `json:"webhookVersion,omitempty"`
```

Comment:
```suggestion
	webhookVersion string `json:"webhookVersion,omitempty"`
```

Reply:
As you mention in https://github.com/kubernetes-sigs/kubebuilder/pull/1869#discussion_r539723800, in the meeting we agreed that the project configuration file should use `webhookVersion` to keep the same string as in the flags to `kb create webhook`. This was mainly because the `webhooks:` in the yaml was in other line so it was a bit easier to see if we kept the full name for the users.

However, the internal representation will always be called like `Resource.Webhooks.Version` which is perfectly clear. Calling it `Resource.Webhooks.WebhookVersion` only adds redundancy. The benefit from the above paragraph doesn't apply to the internal representation. So, unless there is a strong opinion in favor of `Resource.Webhooks.WebhookVersion` from other reviewers/maintainers I would prefer to keep it as `Resource.Webhooks.Version`.

@DirectXMan12 @estroz

Reply:
It is more understandable and intuitive to know that for example `Resource.API.CRDVersion` means :

```yaml
- api:
    crdVersion: v1
    namespaced: true
```

So, IMO do not use in the code the same name that we use to store shows not a good approach at all and can bring confusing. Because of this, I'd advocate in favour of we keep the same nomenclature across the project to represent the same attribute instead of more than one way to call each case.